### ISSUE TITLE
Plan for Volo-HTTP 0.2

### ISSUE THREAD

**Issue by yukiiiteru** [cloudwego/volo#355]
## Big changes

- [x] Refactor `ServerContext` and `Service` on server side
  - #354 
- [x] Remove `State`
  - #354 
- [x] Make client `Request` can be `http::Request` with any `Body`
  - #369
- [x] Use any `Response` and `Error` with `IntoResponse`
  - #361

## Small changes

- [x] Separate mods into parts `client` and `server`
  - #362 
- [x] Add features `client`, `server`, `https`
  - #362 
- [x] Remove `MethodRouterBuilder`
  - #410 

## Features

- [x] Support HTTPS
  - #383 
  - #385 
- [ ] ~Support HTTP/2~
- [x] Support client-side load balance
  - #408 

## Others

- [ ] Add unit tests
- [ ] Add mock support
  - [ ] Service/Layer level: add mock context
  - [ ] Request/Response level: add mock connection and stream
- [x] Add more documents
  - #471

**Comment by wfly1998** [comment]
Break changes are finished, maybe :)

I think we can release a new rc version, and if there is no problem, it can become a stable version.

**Comment by PureWhiteWu** [comment]
Great!
Maybe we can release a new rc version, and publish the stable version after https/http2 is supported?

**Comment by wfly1998** [comment]
There is a important config we are using that is not supported in `hyper-util` currently, so we decide not to support HTTP/2 in Volo-HTTP 0.2, but we will update some function names or variable names for future support of HTTP/2.

**Comment by GuangmingLuo** [comment]
For the last task listed above, please add the docs for http to the websites when it's ready @wfly1998

**Comment by wfly1998** [comment]
> For the last task listed above, please add the docs for http to the websites when it's ready @wfly1998

Of course.  It is expected to be released after the document is completed, document is very important.  But the unit test and mock support may be delayed 😢

**Comment by wfly1998** [comment]
It looks like all breaking changes are complete and we're preparing to add more documentation and release it tomorrow.

Unit tests and mocking support will be updated later.
